src/erlfmt.erl

```diff
@@ -102,6 +108,31 @@ contains_pragma_node({attribute, Meta, _AfAtom, _AbstractExprs}) ->
 contains_pragma_node(_) ->
     false.
 
+%% insert_pragma_nodes only inserts an @format comment,
+%% if one has not already been inserted.
+insert_pragma_nodes([]) -> [];
+insert_pragma_nodes([Node | Nodes]) ->
+    case contains_pragma_node(Node) of
+        true -> [Node | Nodes];
+        false -> [insert_pragma_node(Node) | Nodes]
+    end.
+
+insert_pragma_node({attribute, Meta, AfAtom, AbstractExprs}) ->
```

Comment:
It's not necessarily an attribute as the first node. It could be any expression in something like `rebar.config`. Using `erlfmt_scan:get_anno` and `erlfmt_scan:put_anno` directly on the `Node` should work just fine - there's no need for extracting the `Meta` part manually.

Reply:
I have updated this and added tests for rebar.config and .escript files.  Still working on other comments.